util/UnitEvent.lua

```diff
@@ -0,0 +1,186 @@
+--[[
+    UnitEvent (Namesake)
+        Originally by Bribe
+
+    Uses Magic Defense instead of Defend ability to detect leave.    
+]]
+
+require('CeresStdLib.base.Rawcode')
+require('CeresStdLib.base.Optimizations')
+require('CeresStdLib.base.Native')
+require('CeresStdLib.base.Init')
+require('CeresStdLib.base.Log')
+
+require('CeresStdLib.util.World')
+
+DETECT_LEAVE            = fromRawCode('Amdf')
+
+EVENT_UNIT_ENTER        = 1
+EVENT_UNIT_LEAVE        = 2
+
+UnitEvent = {}
+
+local trigTable         = {}
+local __userData        = {}
+__userData[-1]          = {}   --  Holds references to each unit

```

Comment:
Why not just another local var? At least, why not unique name inside `__userdata`?